src/transformers/modeling_flash_attention_utils.py

```diff
@@ -209,7 +209,7 @@ def fa_peft_integration_check(
     if target_dtype is None:
         return query, key, value
 
-    input_dtype = value.dtype
+    input_dtype = query.dtype
     if input_dtype == torch.float32:
```

Comment:
```suggestion
    if any([module.dtype == torch.float32 for module in [query, key, value]]):
```
might it be better to check all the modules? or is there a performance penalty in doing this?